### DISCUSSION TITLE
Adding additional components to rqlite runtime

### DISCUSSION THREAD

**Discussion by 3goats** [rqlite/rqlite#1039] [Q&A] [answered]
Is it possible to extend rqlite to add additional components to its architecture ? i.e. piggyback rqlite's underlying RAFT subsystem and go translation layer to plugin custom business logic. I hope the question makes sense.

**Comment by otoolep** [comment]
rqite has not been developed with this application in in mind. Of course, as it's open source in principle people can modify rqlite as they see fit.

**Comment by otoolep** [comment, marked as answer]
A shorter answer: no, not without extra programming.
